include/twin_private.h

```diff
@@ -15,11 +15,23 @@
 #define maybe_unused __attribute__((unused))
 
 /*
- * Post-transformed points are stored in 12.4 fixed point
- * values
+ * Fixed-point type definitions
```

Comment:
You can add some materials from Wikipedia: https://en.wikipedia.org/wiki/Fixed-point_arithmetic
Make the comment informative.

In the design of TWIN, the dimension is set to 4096x4096 as the maximum resolution, and the valid range of fixed-point arithmetic is applied accordingly.

Reply:
I find it was mentioned in [TWIN - Geometric Objects](https://keithp.com/~keithp/talks/twin-ols2005/twin-ols2005www/)

>The 12.4 fixed point pixel coordinates provide sufficient resolution to accurately reproduce object geometry on the screen. Note that the screen is therefore implicitly limited to 4096 pixels square.

Based on `#define TWIN_SFIXED_MIN (-0x7fff)`, it can be deduced that the design of this fixed-point type uses the two's complement system. Then why is the limit of the resolution $2^12$? And not $2^11$? One bit out of the 12 bits is used to indicate the sign. 11 bits represent the integer.

Reply:
As for "Make the comment informative," I will refer to the specification ISO/IEC DTR 18037:2008, specifically 4.2 Detailed changes to ISO/IEC 9899:1999, which mentions the new clause 6.2.6.3 - Fixed-point types, and update the comment accordingly.

Reply:
> Based on `#define TWIN_SFIXED_MIN (-0x7fff)`, it can be deduced that the design of this fixed-point type uses the two's complement system. Then why is the limit of the resolution 2^{12}? And not 2^{11}? One bit out of the 12 bits is used to indicate the sign. 11 bits represent the integer.

pow(2, 11) for signed value and pow(2, 12) for unsigned one. The original statement should be clarified though.

Reply:
Do we need to classify fixed-point types as signed or unsigned fixed-point types?

For example,

```c
typedef int16_t twin_fixed_t  /* signed fixed-point*/
typedef uint16_t twin_ufixed_t /* unsigned fixed-point*/

typedef int32_t twin_dfixed_t
typedef uint32_t twin_udfixed_t
```